daemon/cluster/executor/container/adapter.go

```diff
@@ -146,6 +149,54 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 	return nil
 }
 
+// waitNodeAttachments validates that NetworkAttachments exist on this node
+// for every network in use by this task. It blocks until the network
+// attachments are ready, or the context times out. If it returns nil, then the
+// node's network attachments are all there.
+func (c *containerAdapter) waitNodeAttachments(ctx context.Context) error {
+	// to do this, we're going to get the attachment store and try getting the
+	// IP address for each network. if any network comes back not existing,
+	// we'll wait and try again.
+	attachmentStore := c.backend.GetAttachmentStore()
+	if attachmentStore == nil {
+		return fmt.Errorf("error getting attachment store")
+	}
+
+	// TODO(dperny): essentially, we're long-polling here. this is really
+	// sub-optimal, but a better solution based off signaling channels would
+	// require a more substantial rearchitecture.
```

Comment:
Can you open an issue to detail this work?

Reply:
What if I changed it to a `NOTE` or just dropped the qualifier altogether? I don't think the work will really ever be worth it unless we get bored or notice a performance bottleneck here.